### PR TITLE
[Doris on ES] Support escape character

### PR DESCRIPTION
**Pull request by BabySid**
#2864 

add some comments of last commit by the way

### CODE REVIEW

File: be/src/exec/es/es_query_builder.cpp [hunk 1]
```diff
@@ -110,8 +110,17 @@ void WildCardQueryBuilder::to_json(rapidjson::Document* document, rapidjson::Val
 }
 WildCardQueryBuilder::WildCardQueryBuilder(const ExtLikePredicate& like_predicate) : _field(like_predicate.col.name) {
     _like_value = like_predicate.value.to_string();
-    std::replace(_like_value.begin(), _like_value.end(), '_', '?');
-    std::replace(_like_value.begin(), _like_value.end(), '%', '*');
+    // example of translation :
+    //      abc\_123 ===> abc?123
```
Comment (wuyunfeng) [inline be/src/exec/es/es_query_builder.cpp]: How about abc_123 ?

File: be/src/exec/es_http_scan_node.cpp [hunk 2]
```diff
@@ -166,6 +166,11 @@ Status EsHttpScanNode::start_scanners() {
 }
 
 Status EsHttpScanNode::collect_scanners_status() {
+    // NOTE. if open() was called, but set_range() was NOT called for some reason.
```
Comment (wuyunfeng) [inline be/src/exec/es_http_scan_node.cpp]: This comment seems not related with this PR?

**Review by wuyunfeng — COMMENTED** [review]
@blackfox1983  nice work!  I left some comments

File: be/src/exec/es_http_scan_node.cpp (hunk 2 above)
Comment (blackfox1983) [reply be/src/exec/es_http_scan_node.cpp]: Yes. it's last pr i commited. it was committed failed at that time

File: be/src/exec/es/es_query_builder.cpp [hunk 3]
```diff
@@ -110,8 +110,19 @@ void WildCardQueryBuilder::to_json(rapidjson::Document* document, rapidjson::Val
...
     _like_value = like_predicate.value.to_string();
-    std::replace(_like_value.begin(), _like_value.end(), '_', '?');
-    std::replace(_like_value.begin(), _like_value.end(), '%', '*');
+    // example of translation :
+    //      abc_123  ===> abc?123
+    //      abc%ykz  ===> abc*123
+    //      abc\\_123 ===> abc\\_123
+    //      abc\\%123 ===> abc\\%123
+    // NOTE. user must input sql like 'abc\\_123' or 'abc\\%ykz'
+    for (int i = 0; i< _like_value.size(); i++) {
```
Comment (wuyunfeng) [inline be/src/exec/es/es_query_builder.cpp]: Can we remove `#include <boost/algorithm/string/replace.hpp>` in this file?

**Review by wuyunfeng — COMMENTED** [review]
LGTM, I left a minor comment

**Review by wuyunfeng — APPROVED** [review]
@blackfox1983  Thanks

**Review by morningman — APPROVED** [review]
LGTM
